cmake/gen_cmake_skeleton.py

```diff
@@ -276,13 +290,35 @@ def get_test_exe_name(filename):
             ret.append("endif()")
 
         ret.append("""
+if(CONDA_ROOT)
+    if(MSVC)
+        install(TARGETS {tgt}
+            EXPORT kaldi-targets
+            RUNTIME
+                DESTINATION ${{CMAKE_INSTALL_BINDIR}}
+                COMPONENT kaldi
+        )
+
+    else()
+        install(TARGETS {tgt}
+            EXPORT kaldi-targets
+            LIBRARY
+                DESTINATION ${{CMAKE_INSTALL_LIBDIR}}
+                COMPONENT kaldi
+            RUNTIME
+                DESTINATION ${{CMAKE_INSTALL_BINDIR}}
+                COMPONENT kaldi
+        )
+    endif()
+else() # Original functionality
```

Comment:
I don't think you need `if(MSVC)` if you also omit the `DESTINATION` clauses, which are set to their defaults here anyway (and in the `else()` clause below, too. Both .exe and .dll files are installed into the RUNTIME destination on Windows, i.e. shared library targets are RUNTIME, not LIBRARY. Feel free to add `COMPONENT kaldi` to the `install` command below. Then you won't need even the `if(CONDA_ROOT)`.

I am not sure why do you need to define everything as a component. I was thinking of _maybe_ using components eventually for `lib` and `extlib`. Is it required for the Conda build?

Reply:
The components go back to the other comment with the version of OpenFst on conda forge for Mac and Windows.  I used components to install just the kaldi binaries and libraries on those so to not conflict with OpenFst-1.8.2, but on Windows, OpenFst have the `kaldi` component so that they get installed alongside Kaldi.

Reply:
I think our porting currently goes up to 1.7.2 or 1.7.3, we've never moved up to 1.8.2. And Kaldi does not compile with 1.8. The libfst.so by itself is likely to conflict, too. Is it better to tie Kaldi on Conda to the same version of FST that we're normally building against?

We can certainly define `COMPONENT kaldi` in https://github.com/kkm000/openfst, although I would rather prefer something more semantically sounding, for example, `kaldi-rt`, because the component's semantic is "runtime-only dependencies of Kaldi." By the way, I _think_ that besides the main library, the only extension that Kaldi really depends on is far binaries. Possibly the rho and phi matching binaries (I do not remember the component name, and I'm away from a real computer now, too) for the relatively new grammar support, but hardly anything beyond that. Let me double check and respond later. tools/Makefile is not an anchor of trust, it builds more than necessary.

Do you remember off the cuff if CMake can declare a component by purely combining other components?

What looks kinda weird is that the whole of Kaldi is also installed as a single component `kaldi`. Is it necessary?

Reply:
Right, the linking component to libfst I changed here: https://github.com/mmcauliffe/kaldi/blob/master/cmake/third_party/openfst.cmake#L70, so that it links to the versioned `libfst.16.so` (if I'm remembering the version correctly).

Yeah, I think `kaldi-rt` makes sense for OpenFst.  I'm not 100% on how components interact together, I think you'd have to specify each one for installation if you specify any, but the main functionality that's necessary is being able to exclude `kaldi-rt` binaries for installation on mac and linux.